diff --git a/lib/gollum/templates/layout.mustache b/lib/gollum/templates/layout.mustache
index 6e941f61..a134253c 100644
--- a/lib/gollum/templates/layout.mustache
+++ b/lib/gollum/templates/layout.mustache
@@ -58,6 +58,8 @@
   <title>{{title}}</title>
 </head>
 <body>
+{{< user}}
+
 <div class="container-lg clearfix">
 {{{yield}}}
 </div>
diff --git a/lib/gollum/templates/user.mustache b/lib/gollum/templates/user.mustache
new file mode 100644
index 00000000..39ce12bf
--- /dev/null
+++ b/lib/gollum/templates/user.mustache
@@ -0,0 +1,15 @@
+<div id="user" style="
+    text-align: right;
+    padding-right:0.5em;
+    line-height: 2.0em;
+    color: #666666;
+">
+     <p>
+     {{#user_authed}}
+       {{user_name}} ({{user_email}}) | <strong><a href="/oauth2/sign_out">[Logout]</a></strong>
+     {{/user_authed}}
+     <p>
+ </div>
diff --git a/lib/gollum/views/layout.rb b/lib/gollum/views/layout.rb
index c70d6b40..337c94fa 100644
--- a/lib/gollum/views/layout.rb
+++ b/lib/gollum/views/layout.rb
@@ -74,7 +74,19 @@ module Precious
       def history
         false
       end
-      
+
+      def user_authed
+        @user_authed
+      end
+
+      def user_email
+        @user.email
+      end
+
+      def user_name
+        @user.name
+      end
+
       def overview
         false
       end
