diff --git a/lib/gollum/templates/layout.mustache b/lib/gollum/templates/layout.mustache
index 0cb7b1e1..3982131f 100644
--- a/lib/gollum/templates/layout.mustache
+++ b/lib/gollum/templates/layout.mustache
@@ -133,6 +133,7 @@
   <title>{{title}}</title>
 </head>
 <body>
+{{< user}}
 <div class="container-lg clearfix">
 {{{yield}}}
 </div>
diff --git a/lib/gollum/views/layout.rb b/lib/gollum/views/layout.rb
index 5acb983c..68c2912d 100644
--- a/lib/gollum/views/layout.rb
+++ b/lib/gollum/views/layout.rb
@@ -90,6 +90,14 @@ module Precious
         false
       end

+      def user_email
+        session["gollum.author"]["email"]
+      end
+
+      def user_name
+        session["gollum.author"]["name"]
+      end
+
       def overview
         false
       end
diff --git a/lib/gollum/views/overview.rb b/lib/gollum/views/overview.rb
index f2d816ec..873d040a 100644
--- a/lib/gollum/views/overview.rb
+++ b/lib/gollum/views/overview.rb
@@ -75,6 +75,9 @@ module Precious
         true
       end

+      def search
+        true
+      end

     end
   end
