diff --git a/lib/gollum/templates/layout.mustache b/lib/gollum/templates/layout.mustache
index 8864be69..942d9cdf 100644
--- a/lib/gollum/templates/layout.mustache
+++ b/lib/gollum/templates/layout.mustache
@@ -58,6 +58,7 @@
   <title>{{title}}</title>
 </head>
 <body>
+{{< user}}
 <div class="container-lg clearfix">
 {{{yield}}}
 </div>
diff --git a/lib/gollum/templates/user.mustache b/lib/gollum/templates/user.mustache
new file mode 100644
index 00000000..ca964f93
--- /dev/null
+++ b/lib/gollum/templates/user.mustache
@@ -0,0 +1,10 @@
+<div id="user" style="
+    text-align: right;
+    padding-right:0.5em;
+    line-height: 2.0em;
+    color: #666666;
+">
+    <p>
+        {{user_name}} ({{user_email}}) | <strong><a href="/oauth2/sign_out">[Logout]</a></strong>
+    <p>
+</div>
diff --git a/lib/gollum/views/helpers/locale_helpers.rb b/lib/gollum/views/helpers/locale_helpers.rb
index da15bc76..e72add4e 100644
--- a/lib/gollum/views/helpers/locale_helpers.rb
+++ b/lib/gollum/views/helpers/locale_helpers.rb
@@ -77,6 +77,14 @@ module Precious
       def no_method_message(method_name, message = NO_METHOD_MESSAGE)
         "[#{message}: #{method_name}]"
       end
+
+      def current_user_email
+        session["gollum.author"]["email"]
+      end
+
+      def current_user_name
+        session["gollum.author"]["name"]
+      end
     end
   end
 end
diff --git a/lib/gollum/views/layout.rb b/lib/gollum/views/layout.rb
index c1f825f0..21132493 100644
--- a/lib/gollum/views/layout.rb
+++ b/lib/gollum/views/layout.rb
@@ -90,6 +90,14 @@ module Precious
         false
       end
       
+      def current_user_email
+        @current_user_email
+      end
+
+      def current_user_name
+        @current_user_name
+      end
+
       def overview
         false
       end
diff --git a/lib/gollum/views/overview.rb b/lib/gollum/views/overview.rb
index 0c651296..518f1da9 100644
--- a/lib/gollum/views/overview.rb
+++ b/lib/gollum/views/overview.rb
@@ -74,8 +74,9 @@ module Precious
       def latest_changes
         true
       end
-    
-      
+      def search
+        true
+      end
     end
   end
 end
