diff --git a/lib/gollum/app.rb b/lib/gollum/app.rb
index 4e7c3ea8..d62f7cb3 100644
--- a/lib/gollum/app.rb
+++ b/lib/gollum/app.rb
@@ -143,6 +143,7 @@ module Precious
       forbid unless @allow_editing || request.request_method == 'GET'
     end
 
+
     get '/' do
       redirect clean_url(::File.join(@base_url, wiki_new.index_page))
     end
diff --git a/lib/gollum/templates/layout.mustache b/lib/gollum/templates/layout.mustache
index 8864be69..942d9cdf 100644
--- a/lib/gollum/templates/layout.mustache
+++ b/lib/gollum/templates/layout.mustache
@@ -58,6 +58,7 @@
   <title>{{title}}</title>
 </head>
 <body>
+{{< user}}
 <div class="container-lg clearfix">
 {{{yield}}}
 </div>
diff --git a/lib/gollum/templates/user.mustache b/lib/gollum/templates/user.mustache
new file mode 100644
index 00000000..ed6701aa
--- /dev/null
+++ b/lib/gollum/templates/user.mustache
@@ -0,0 +1,10 @@
+<div id="user" style="
+    text-align: right;
+    padding-right:0.5em;
+    line-height: 2.0em;
+    color: #666666;
+">
+    <p>
+        {{gollum_author_email}} | <strong><a href="/oauth2/sign_out">[Logout]</a></strong>
+    <p>
+</div>
diff --git a/lib/gollum/views/layout.rb b/lib/gollum/views/layout.rb
index c1f825f0..4e542e20 100644
--- a/lib/gollum/views/layout.rb
+++ b/lib/gollum/views/layout.rb
@@ -90,6 +90,10 @@ module Precious
         false
       end
       
+      def gollum_author_email
+        @gollum_author_email
+      end
+
       def overview
         false
       end
diff --git a/lib/gollum/views/overview.rb b/lib/gollum/views/overview.rb
index 0c651296..518f1da9 100644
--- a/lib/gollum/views/overview.rb
+++ b/lib/gollum/views/overview.rb
@@ -74,8 +74,9 @@ module Precious
       def latest_changes
         true
       end
-    
-      
+      def search
+        true
+      end
     end
   end
 end
