diff --git a/lib/gollum/templates/layout.mustache b/lib/gollum/templates/layout.mustache
index 8864be69..942d9cdf 100644
--- a/lib/gollum/templates/layout.mustache
+++ b/lib/gollum/templates/layout.mustache
@@ -58,6 +58,7 @@
   <title>{{title}}</title>
 </head>
 <body>
+{{< user}}
 <div class="container-lg clearfix">
 {{{yield}}}
 </div>
diff --git a/lib/gollum/templates/user.mustache b/lib/gollum/templates/user.mustache
new file mode 100644
index 00000000..10fdafa4
--- /dev/null
+++ b/lib/gollum/templates/user.mustache
@@ -0,0 +1,12 @@
+<div id="user" style="
+    text-align: right;
+    padding-right:0.5em;
+    line-height: 2.0em;
+    color: #666666;
+">
+    <p>
+        {{#gollum_author_email}}
+            {{gollum_author_email}} | <strong><a href="{{gollum_logout_link}}">[Logout]</a></strong>
+        {{/gollum_author_email}}
+    <p>
+</div>
diff --git a/lib/gollum/views/layout.rb b/lib/gollum/views/layout.rb
index c1f825f0..d0dff9c6 100644
--- a/lib/gollum/views/layout.rb
+++ b/lib/gollum/views/layout.rb
@@ -90,6 +90,14 @@ module Precious
         false
       end
       
+      def gollum_author_email
+        @gollum_author_email
+      end
+
+      def gollum_logout_link
+        @gollum_logout_link
+      end
+
       def overview
         false
       end
